.github/workflows/ci.yml

```diff
@@ -388,6 +388,74 @@ jobs:
       run: cmake --build build --target test_cmake_build
 
 
+  # Testing on ICX/ICPX, Intel's next-gen C/C++ compiler using the oneAPI apt repo
+  icx:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+
+    name: "🐍 3 • ICPX latest • x64"
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Add apt repo
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y wget build-essential pkg-config cmake ca-certificates gnupg
+        wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+        sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB
+        echo "deb https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
+    - name: Add ICC & Python 3
+      run: sudo apt-get update; sudo apt-get install -y intel-oneapi-compiler-dpcpp-cpp cmake python3-dev python3-numpy python3-pytest python3-pip
+
+    - name: Update pip
+      shell: bash
+      run: |
+        set +e; source /opt/intel/oneapi/setvars.sh; set -e
+        python3 -m pip install --upgrade pip
+    - name: Install dependencies
+        set +e; source /opt/intel/oneapi/setvars.sh; set -e
```

Comment:
Found it!

```suggestion
      run: |
        set +e; source /opt/intel/oneapi/setvars.sh; set -e
```

Reply:
Wuhu, thank you for spotting this! :sparkling_heart:

Reply:
Thanks for taking care of this PR :-) (And a fresh outsider glance at code does miracles ;-) )